### DISCUSSION TITLE
hopefully simple Newbie question about playlists

### DISCUSSION THREAD

**Discussion by ColmOsiris** [quodlibet/quodlibet#3754] [Q&A] [answered]
hi everyone! i've been using a Mac-based audio player called Swinsian. my needs are very simple really. it does everything i want, except for Replay Gain. i need one playlist, which will have about 5,000 songs in it, which i want Replay Gain for, and which i need to select by genre etc, and a group of other playlists into which i will put albums as and when i want to play them.

Quod Libet seems to be a very sophisticated app, and i'm having difficulty navigating the User Guide for my simple needs. partly, this is to do with terminology, i think. for instance, i can't figure out what the difference is between Queues and Song Lists, or whether it's something i even need to be concerned about. i can see that QL does a huge number of things i'm not likely to need in the short term, though that may change.

so really, i just need to know how to set up the above playlists, and how to select them. i have a list of Artists, Albums, and Songs, but not playlists. what i can see on my GUI is not the same as the screenshots in the User Guide. and also, if i have the player on shuffle, how can i see the upcoming list of songs that will play?

thanks for reading.

**Comment by zeezhao** [comment, marked as answer]
Hi. As mentioned QL has loads of stuff....

What you can do (and I am using it on a Windows machine) is open it and set via "Browse/Playlists". This will have Playlists as your main window. Then create a playlist using "+" New sign on bottom left.

You can then open an additional window via "Browse/Open Browser/Track List". This is handy for searching and adding to a Playlist. Right click on any track or highlighted tracks, then add to required playlist via "Playlists" menu. (You may have to click on the playlist again to get it to refresh newly added tracks...)

Replay Gain can be set via "File/Preferences/Playback", see the "Replay Gain Volume Adjustment" section.

Hope this helps.

**Reply by ColmOsiris** [reply]
really sorry for the delay. thank you very much for that, yes, that's brilliant. i'm sure i'll get to know more of its features as i go along, but that's a great start. :)

**Reply by ColmOsiris** [reply]
**how to delete stuff?**

sorry, spoke a bit too soon!

i started adding tracks to a playlist, but i mistakenly added some twice. i can't find any way of deleting a playlist, or of deleting individual or ranges of tracks.

i looked at the Help, and did a search, but nothing that came up in the "delete playlist" search included "delete", only "playlist". and a search for "delete" gave only two results, neither of which were helpful.

i even tried deleting the app and its plist and reinstalling it, but guess what! the playlist and the tracks are still there!

i didn't want to come back here so soon, but... H--E--L--P!

**Reply by zeezhao** [reply]
Great! 
To delete from playlist, just right-click on track and press "Remove from Playlist".

To delete Playlist, do same - right-click on it, and press "Delete".

**Reply by zeezhao** [reply]
Note that for instance, on Windows, playlists are stored in:
C:\Users\"user_name"\AppData\Roaming\Quod Libet\playlists

So you can manually remove the file from there.

**Reply by ColmOsiris** [reply]
thank you. the problem seems to be a Mac one, then. right-clicking (or ctrl-clicking) does nothing at all.

on a Mac it's  ~\Library\Preferences\io.github.quodlibet.QuodLibet.plist, which was deleted. yet still the tracks remain.

i've created another playlist for it, but the trouble is i now have duplicate tracks in my library, and no way to get rid of them.

**Reply by zeezhao** [reply]
You'd have to close the app first before deleting the playlist files in that directory.

I have not used it on mac, so don't know differences

Try the "delete" button on keyboard when track is selected...

**Reply by ColmOsiris** [reply]
ok, thanks. i'd better ask a new question about it then. thank you for your help.

**Reply by zeezhao** [reply]
No problem. Also note that I am running version 4.4.0. Please confirm you are running the same version.

**Reply by ColmOsiris** [reply]
thank you. i am, yes. and my OS is 10.12.6 (Sierra). i have a MacBook Pro, which is more recent. i will try it on there as well, though it's on this machine i'd really like it to work.

pressing the delete key while a track is selected does nothing, sadly. pressing ctrl, alt or cmd at the same time as the delete key doesn't do anything either.

**Reply by zeezhao** [reply]
It seems to be a mac issue. See this:
https://github.com/quodlibet/quodlibet/issues/3338

Probably a good idea to see the issues queue for any other Mac issues.

https://github.com/quodlibet/quodlibet/issues?q=is%3Aissue+is%3Aopen+mac

**Reply by ColmOsiris** [reply]
i'm sorry, i thought i'd acknowledged this. yes, it looks like this app is no use for Mac users, which is a great shame. oh well!

**Reply by declension** [reply]
:information_source: To help clarify a few things: **never** delete `.plist` files, they are [MacOS app metadata](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Introduction/Introduction.html) and nothing to do with Quod Libet internals or playlists.

Playlists are (now) [XSPF](https://xspf.org/) files in a directory that depends on your OS / system (in *nix-esque systems typically `~/.quodlibet/playlists`). You _can_ delete these when the app isn't running if you want, or can't use the UI for this (e.g. weird UI bugs as above).

Hope that helps
